### DISCUSSION TITLE
Is there a way to display a glyph by its name?

### DISCUSSION THREAD

**Discussion by Mark2Mark** [coldtype/coldtype#131] [General]
For instance I want to showa glyph from its name "adieresis" instead of "ä". Is that possible? I found some things in the source like `drawGlyphToPen("adieresis", pen)` or  `_getGlyphDrawing` but those seem for internal handling and I can’t see how to use them properly.

**Comment by stenson** [comment]
Definitely possible, though the way I'd do it isn't super coldtype-specific, apart from the `P.glyph` method:

```python
from coldtype import *

mistral = Font.Find("MistralD")
glyphSet = mistral.font.ttFont.getGlyphSet()

@renderable()
def by_glyph(r):
    return (P().glyph(glyphSet["adieresis"], glyphSet)
        .align(r)
        .f(0))
```

Does that work / make sense for your use-case?

**Reply by Mark2Mark** [reply]
Ohhh, I was so close :D
I came so far:
```py
allGlyphs = fnt.font.ttFont.getGlyphOrder()
glyphSet = fnt.font.ttFont.getGlyphSet()
glyphByIndex = glyphSet[allGlyphs[2]]
glyphByName = glyphSet["b"]
```

just didn’t know about the `P().glyph(...` part. That’s great, now I can use both options, either by name, or by index. Thank you so much, Rob!

**Reply by Mark2Mark** [reply]
Actually, by index only works like written, but somehow fails when I add it as `glyphSet[allGlyphs[2]]` to the `.glyph()` method with a key error. Same thing works though, when I add the previously declared variable `glyphByIndex ` into the `.glyph()` method. That could be my bad though, will check tomorrow morning with a fresh brain.
```py
# with the 4 lines from my previous comment
@renderable()
def by_glyph2(r):
    return (P().glyph(glyphByIndex) # Works
        .align(r)
        .f(0))

@renderable()
def by_glyph3(r):
	return (P().glyph(glyphSet[allGlyphs[2]]) # Fails
		.align(r)
		.f(0))
```


Another question: can one apply font attributes to that glyph (especially the font variation values)?

**Reply by stenson** [reply]
Not quite sure on the index issue, but if you switch to using a cmap you could do standard StSt calls with full variation, etc., something like this:

```python
from coldtype import *

font = Font.ColdtypeObviously()

cmap = {v:chr(k) for k,v in font.font.ttFont["cmap"].getBestCmap().items()}

@renderable()
def by_glyph(r):
    return (StSt(cmap["C"], font, 500, wdth=0)
        .align(r))
```
